cms/release_calendar/models.py

```diff
@@ -0,0 +1,207 @@
+from typing import TYPE_CHECKING, Any, ClassVar
+
+from django.conf import settings
+from django.db import models
+from django.utils.functional import cached_property
+from django.utils.text import slugify
+from django.utils.translation import gettext_lazy as _
+from modelcluster.fields import ParentalKey
+from wagtail.admin.panels import FieldPanel, FieldRowPanel, InlinePanel, MultiFieldPanel
+from wagtail.fields import RichTextField
+from wagtail.models import Orderable, Page
+
+from cms.core.fields import StreamField
+from cms.core.models import BasePage, LinkFieldsMixin
+
+from .blocks import (
+    ReleaseCalendarChangesStoryBlock,
+    ReleaseCalendarPreReleaseAccessStoryBlock,
+    ReleaseCalendarStoryBlock,
+)
+from .enums import NON_PROVISIONAL_STATUSES, ReleaseStatus
+from .forms import ReleaseCalendarPageAdminForm
+
+if TYPE_CHECKING:
+    from django.http import HttpRequest
+
+
+class ReleaseCalendarIndex(BasePage):  # type: ignore[django-manager-missing]
+    """The release calendar index page placeholder."""
+
+    template = "templates/pages/release_index.html"
+
+    parent_page_types: ClassVar[list[str]] = ["home.HomePage"]
+    subpage_types: ClassVar[list[str]] = ["ReleaseCalendarPage"]
+    max_count_per_parent = 1
+
+
+class ReleasePageRelatedLink(LinkFieldsMixin, Orderable):
+    """Related links. e.g. https://www.ons.gov.uk/releases/welshlanguagecensus2021inwales."""
+
+    parent = ParentalKey("ReleaseCalendarPage", related_name="related_links", on_delete=models.CASCADE)
+
+
+class ReleaseCalendarPage(BasePage):  # type: ignore[django-manager-missing]
+    """The calendar release page model."""
+
+    base_form_class = ReleaseCalendarPageAdminForm
+    template = "templates/pages/release_calendar/release_calendar_page.html"
+    parent_page_types: ClassVar[list[str]] = ["ReleaseCalendarIndex"]
+    subpage_types: ClassVar[list[str]] = []
+
+    # Fields
+    status = models.CharField(choices=ReleaseStatus.choices, default=ReleaseStatus.PROVISIONAL, max_length=32)
+    summary = RichTextField(features=settings.RICH_TEXT_BASIC)
+
+    release_date = models.DateTimeField(
+        blank=True, null=True, help_text=_("Required once the release has been confirmed.")
+    )
+    release_date_text = models.CharField(
+        max_length=50, blank=True, help_text=_("Format: 'Month YYYY', or 'Month YYYY to Month YYYY'.")
+    )
+    next_release_date = models.DateTimeField(blank=True, null=True)
+    next_release_text = models.CharField(
+        max_length=255, blank=True, help_text=_("Formats: 'DD Month YYYY Time' or 'To be confirmed'.")
+    )
```

Comment:
Non-blocker: If we have both date types, it makes sense to enforce that the next release date much be greater than release date.